### PR TITLE
Add Dockerfile to build project

### PR DESCRIPTION
**Pull request by RamLavi**
This PR introduces the Dockerfile that will build the project.

pushed on top of #1

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM registry.access.redhat.com/ubi9/ubi-minimal
```
Comment (orelmisan) [inline Dockerfile:1]: Please consider adding a tag to the base image.
With a tag, you make sure that you'll get the same base image on every build (thus the same software versions).
This makes the build more predictable.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+GO_IMAGE_TAG := 1.19
+BIN_DIR = $(CURDIR)/_output/bin
+CRI_BIN ?= $(shell hack/detect_cri.sh)
+
+build:
+	$(CRI_BIN) run --rm \
+	           --volume `pwd`:$(CURDIR):Z \
+	           --workdir $(CURDIR) \
+	           --user $(shell id -u):$(shell id -g) \
+	           -e XDG_CACHE_HOME=/tmp/.cache \
```
Comment (orelmisan) [inline Makefile:16]: Why are these necessary?

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Please see the inline comments.

Is there a need for the empty `.dockerignore` file?

File: Dockerfile (hunk 1 above)
Comment (RamLavi) [reply Dockerfile:1]: I see the benefit  you mention, but it also makes us "stuck" in the past and should anything break we would probably not know about it. 
I prefer to know as soon as things break.
This practice is also used on other projects in kubevirt org such as [kubevirt/kubesecondarydns](https://github.com/kubevirt/kubesecondarydns), [kubevirt/bridge-marker](https://github.com/kubevirt/bridge-marker), [kubevirt/ssp-operator](https://github.com/kubevirt/ssp-operator) etc..

File: Makefile (hunk 2 above)
Comment (RamLavi) [reply Makefile:16]: running with docker results with the output folder having root permissions.
In order to fix that I added a specific --user parameters that levels the permissions to the one running the build.
But by doing that I ran into a different problem - docker build is creating a cache folder on `/` folder and that is forbidden for a non-root user. the fix is to set `XDG_CACHE_HOME` to a path that does not require root write permissions.
I looks a bit messy but I think it's worth the trouble in order to have the output folder non-root

**Comment by RamLavi** [comment]
> Is there a need for the empty `.dockerignore` file?

I plan to add content in future PRs (for example .github, tests folders), but right now it seems the right PR to insert it generally.

File: Dockerfile (hunk 1 above)
Comment (orelmisan) [reply Dockerfile:1]: This is good for going forward, but what happens when you need to fix an old version?
The old code with the fix will be built and copied on to a new base image than the one it previously used.

**Comment by orelmisan** [comment]
> > Is there a need for the empty `.dockerignore` file?
> 
> I plan to add content in future PRs (for example .github, tests, infra folders), but right now it seems the right PR to insert it generally.

I suggest to add it when there is a need for it.
If you will forget about it, it will stay empty and confuse future readers.

File: Dockerfile (hunk 1 above)
Comment (RamLavi) [reply Dockerfile:1]: good point, but as this application does not use unorthodox features from this image base, I think it's safe to assume it's rare enough to ignore. 
But for the off chance, I'll add the ability to choose the tag as an env var so that our future selves would have an easier life using it.

**Comment by RamLavi** [comment]
> > > Is there a need for the empty `.dockerignore` file?
> > 
> > 
> > I plan to add content in future PRs (for example .github, tests, infra folders), but right now it seems the right PR to insert it generally.
> 
> I suggest to add it when there is a need for it. If you will forget about it, it will stay empty and confuse future readers.



> > > Is there a need for the empty `.dockerignore` file?
> > 
> > 
> > I plan to add content in future PRs (for example .github, tests, infra folders), but right now it seems the right PR to insert it generally.
> 
> I suggest to add it when there is a need for it. If you will forget about it, it will stay empty and confuse future readers.

I understand you concern but I disagree, Adding the ignore file now will ensure that we have it in place when we do have content that needs to be ignored, especially when I know I will add it in the near future.

File: Makefile (hunk 2 above)
Comment (RamLavi) [reply Makefile:17]: Added [here](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/6fb2fa3018a782b5a5c376eb66b55743d9f440bb..ed3f3d9cdec38970c06d397cbd8c528b858e9c88)

**Review by orelmisan — APPROVED** [review]
Thank you for the changes.

File: Dockerfile (hunk 1 above)
Comment (RamLavi) [reply Dockerfile]: Added [here](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/6fb2fa3018a782b5a5c376eb66b55743d9f440bb..ed3f3d9cdec38970c06d397cbd8c528b858e9c88)
